graveler/committed/sstable/writer.go

```diff
@@ -0,0 +1,85 @@
+package sstable
+
+import (
+	"fmt"
+	"hash"
+
+	"github.com/cockroachdb/pebble/sstable"
+	"github.com/treeverse/lakefs/graveler"
+	"github.com/treeverse/lakefs/pyramid"
+)
+
+type DiskWriter struct {
+	w      *sstable.Writer
+	tierFS pyramid.FS
+
+	first graveler.Key
+	last  graveler.Key
+	count int
+	hash  hash.Hash
+
+	fh *pyramid.File
+}
+
+func newDiskWriter(tierFS pyramid.FS, hash hash.Hash) (*DiskWriter, error) {
+	fh, err := tierFS.Create(sstableTierFSNamespace)
+	if err != nil {
+		return nil, fmt.Errorf("opening file: %w", err)
+	}
+
+	writer := sstable.NewWriter(fh, sstable.WriterOptions{
+		Compression: sstable.SnappyCompression,
+	})
+
+	return &DiskWriter{
+		w:      writer,
+		fh:     fh,
+		tierFS: tierFS,
+		hash:   hash,
+	}, nil
+}
+
+func (dw *DiskWriter) WriteRecord(record graveler.ValueRecord) error {
+	keyBytes := []byte(record.Key)
+	valBytes, err := serializeValue(*record.Value)
+	if err != nil {
+		return fmt.Errorf("serializing entry: %w", err)
+	}
+	if err := dw.w.Set(keyBytes, valBytes); err != nil {
+		return fmt.Errorf("setting key and value: %w", err)
+	}
+
+	// updating stats
+	if dw.count == 0 {
+		dw.first = record.Key
+	}
+	dw.last = record.Identity
+	dw.count++
+
+	if _, err := dw.hash.Write(keyBytes); err != nil {
```

Comment:
I don't think this hashing works.  Specifically, it will hash `keyBytes, valueBytes = []byte{1,2,3}, []byte{4,5}` to the same value as `keyBytes, valueBytes = []byte{1,2},[]byte{3,4,5}`.  Hashing is like marshalling (serialization): you need to write things that you _could_ unmarshal, had the hash only not clobbered them.  So e.g. first write `len(keyBytes)` (in you favourite binary encoding) and then `keyBytes`.  Or choose a separator and escape it out of `keyBytes`.  Or pass `dw.hash` to some existing marshalling library.